### ISSUE TITLE
VmRemovedEvent for a VM Template has template field set to False

### ISSUE THREAD

**Issue by tim-ireland** [vmware/pyvmomi#400]
Hi, question regarding eventing, specifically VM Events.  

I'm working on processing some events from vSphere using vSphere Version 6.0.0 Build 3271482 and pyVmomi version 6.0.0 with python 2.7.11.  In vSphere I have created a VM template by cloning a VM to a Template.  If I then Delete the template, by right clicking and selecting 'Delete From disk' I then receive a `VmRemovedEvent`, as seen below.

The issue is that I need to distinguish between events for VM's and events for VM templates and most events for VM templates have `template = true` set in the event, but for `VmRemovedEvent` I see `template = false`.  Is this expected?

``` python
   (vim.event.VmRemovedEvent) {
      dynamicType = <unset>,
      dynamicProperty = (vmodl.DynamicProperty) [],
      key = 3765733,
      chainId = 3765731,
      createdTime = 2016-05-23T19:34:00.687Z,
      userName = 'admin',
      datacenter = (vim.event.DatacenterEventArgument) {
         dynamicType = <unset>,
         dynamicProperty = (vmodl.DynamicProperty) [],
         name = 'Switch1',
         datacenter = 'vim.Datacenter:datacenter-2'
      },
      computeResource = (vim.event.ComputeResourceEventArgument) {
         dynamicType = <unset>,
         dynamicProperty = (vmodl.DynamicProperty) [],
         name = 'Dell Cluster',
         computeResource = 'vim.ClusterComputeResource:domain-c7'
      },
      host = (vim.event.HostEventArgument) {
         dynamicType = <unset>,
         dynamicProperty = (vmodl.DynamicProperty) [],
         name = 'esxi-1',
         host = 'vim.HostSystem:host-534'
      },
      vm = (vim.event.VmEventArgument) {
         dynamicType = <unset>,
         dynamicProperty = (vmodl.DynamicProperty) [],
         name = 'VMTemplate',
         vm = 'vim.VirtualMachine:vm-575'
      },
      ds = <unset>,
      net = <unset>,
      dvs = <unset>,
      fullFormattedMessage = 'Removed VMTemplate on esxi-1 from Switch1',
      changeTag = <unset>,
      template = false
   }
```

**Comment by tianhao64** [comment]
Looks like a vsphere bug to me. What are the other events you tried?

**Comment by tim-ireland** [comment]
Thanks @tianhao64, I've tried the following events related to VM Templates.  Items I believe to be incorrect are in **bold**.

1. VM converted to template: 
    Observed `VmResourcePoolMovedEvent` with `template = true`

2. VM cloned to template:
    **! Observed `VmRemovedEvent` with `template = false`**
    Observed `VmResourcePoolMovedEvent` with `template = true`

3. VM cloned to Virtual Machine:
    Observed `VmClonedEvent` with `template = false`

4. VM Template converted to Virtual Machine:
    **! No VmEvent seen**

5. New VM from Template:
    Observed `VmDeployedEvent` with `template = false`

6. VM Template cloned to Template:
    **! Observed `VmClonedEvent` with `template = false`**
    Observed `VmResourcePoolMovedEvent` with `template = true`

7. Delete a VM Template
    **! Observed `VmRemovedEvent` with `template = false`**

**Comment by tim-ireland** [comment]
A workaround to get the template status of a VM given a `VmClonedEvent` or `VmRemoveEvent` is to check the `event.vm.vm.config.template` value.

**Comment by tianhao64** [comment]
@tim-ireland Thanks for the detailed response. I will double check and file bug against vsphere if they are real bugs.

**Comment by tim-ireland** [comment]
Just to follow up, `event.vm.vm.config.template` does not seem to be available on `VmRemoved` event, so the workaround only works for `VmClonedEvent`.

**Comment by tianhao64** [comment]
@tim-ireland I have filed a bug against vsphere.

**Comment by DanielDraganov** [comment]
Fixed in vSphere 6.5
